hooks/multi_publish/nuke/secondary_publish.py

```diff
@@ -656,8 +657,11 @@ def _update_flame_clip(self, clip_path, write_node, sg_publish, progress_cb):
         # and lastly plug in the values
         render_path_fields["SEQ"] = format_str % (min_frame, max_frame) 
         
-        # contruct the final path             
-        publish_path_flame = publish_template.apply_fields(render_path_fields)
+        # contruct the final path - because flame doesn't have any windows support and 
+        # because the "hub" platform is always linux (with potential flame assist and flare
+        # satellite setups on macosx), request that the paths are written out on linux form
+        # regardless of the operating system currently running.
+        publish_path_flame = publish_template.apply_fields(render_path_fields, TemplatePath.PLATFORM_LINUX)
```

Comment:
I totally agree - I was really unsure about this particular detail - I started out doing that but it felt a little odd - in fact, I ended up writing this seemingly simple code change three of four times because I wasn't sure what was right. It seemed like wiring it up to `sys.platform` meant we were married to those constants, making it tricky to for example upgrade client to python3 (returning linux). But the main concern was that we may extend our concept of what a platform is to include other things beyond operating systems. It would be cool if it could return cloud based things in the future, or things like perforce or dropbox paths. A minor detail was that "win32, darwin and linux2" all feel really technical.

```
templ.apply_fields(fields, TemplatePath.PLATFORM_SHOTGUN)
templ.apply_fields(fields, TemplatePath.PLATFORM_URL)
templ.apply_fields(fields, TemplatePath.PLATFORM_PERFORCE)
templ.apply_fields(fields, TemplatePath.PLATFORM_DROPBOX)
templ.apply_fields(fields, TemplatePath.PLATFORM_CLOUD)
```

Rather speculative, I know and I am not a big fan of speculation - in this particular case, however, it seemed like weighing up the pros and cons, I ended up with the separate enum. But only after trying the other approach first. I am pretty hesitant about it. Super curious to hear what you think.